packages/@webex/plugin-cc/src/AgentProfileService/AgentProfileService.ts

```diff
@@ -0,0 +1,146 @@
+import {WebexSDK, HTTP_METHODS} from '../types';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from './types';
+
+export default class AgentProfileService {
+  ciUserId: string;
+  orgId: string;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, orgId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.orgId = orgId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get User using CI.
+   * @param {string} ciUserId The CI Id of the user.
+   * @param {string} orgId The ID of an Organization.
+   * @returns {Promise<UserResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getUserUsingCI(ciUserId: string, orgId: string): Promise<UserResponse> {
+    try {
+      if (!ciUserId || !orgId) Promise.reject(new Error('ciUserId or orgId is undefined'));
+
+      const URL = `${this.wccAPIURL}organization/${orgId}/user/by-ci-user-id/${ciUserId}`;
+      this.webex.logger.log('getUserUsingCI api called successfully.');
+
+      return Promise.resolve(await this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling getUserUsingCI API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get Desktop Profile by passing Org Id.
+   * @param {string} orgId The ID of an Organization.
+   * @param {string} desktopProfileId ID of the Desktop Profile to be retrieved.
+   * @returns {Promise<DesktopProfileResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async retrieveDesktopProfileById(
+    orgId: string,
+    desktopProfileId: string
+  ): Promise<DesktopProfileResponse> {
+    try {
+      if (!orgId || !desktopProfileId)
```

Comment:
Do we really need this condition here? Both orgId and desktopProfileId are mandatory arguments. In what scenario would they not be passed correctly?

Reply:
Code Corrected.